source/material_model/viscoelastic_plastic.cc

```diff
@@ -0,0 +1,796 @@
+                                        / ( std::sqrt(3.0) * (3.0 + std::sin(phi[j]) ) )
+                                        :
+                                        coh[j] * std::cos(phi[j]) + std::max(pressure,0.0) * std::sin(phi[j]) );
+
+                  if ( stresses_viscous[j] >= stresses_yield[j]  )
+                    {
+                      viscosities_drucker_prager[j] = stresses_yield[j] / (2.0 * edot_ii);
+                    }
+                  else
+                    {
+                      viscosities_drucker_prager[j] = viscosities_pre_yield[j];
+                    }
+
+                  viscosities_yield[j] = viscosities_drucker_prager[j];
+
+                  viscosities_viscoelastic[j] = ( viscosities_yield[j] * dte ) / ( dte + ( viscosities_yield[j] / elastic_shear_moduli[j] ) );
+
+                }
+
+              out.viscosities[i] = average_value(volume_fractions,viscosities_viscoelastic,viscosity_averaging);
+
+              if (ElasticAdditionalOutputs<dim> *elastic_out = out.template get_additional_output<ElasticAdditionalOutputs<dim> >())
+                {
+                  elastic_out->elastic_shear_moduli[i] = average_value(volume_fractions,elastic_shear_moduli,viscosity_averaging);
+                }
+
+              // Fill elastic force outputs (assumed to be zero during intial time step)
+              if (force_out)
+                {
+                  force_out->elastic_force[i] = 0.;
+                }
+            }
+        }
+
+      // Viscoelasticity section
+      if (in.current_cell.state() == IteratorState::valid && this->get_timestep_number() > 0 && in.strain_rate.size() > 0)
+        {
+          // Get old (previous time step) velocity gradients
+          std::vector<Point<dim> > quadrature_positions(in.position.size());
+          for (unsigned int i=0; i < in.position.size(); ++i)
+            quadrature_positions[i] = this->get_mapping().transform_real_to_unit_cell(in.current_cell, in.position[i]);
+
+          // FEValues requires a quadrature and we provide the default quadrature
+          // as we only need to evaluate the solution and gradients.
+          FEValues<dim> fe_values (this->get_mapping(),
+                                   this->get_fe(),
+                                   Quadrature<dim>(quadrature_positions),
+                                   update_gradients);
+
+          fe_values.reinit (in.current_cell);
+          std::vector<Tensor<2,dim> > old_velocity_gradients (quadrature_positions.size(), Tensor<2,dim>());
+          fe_values[this->introspection().extractors.velocities].get_function_gradients (this->get_old_solution(),
+                                                                                         old_velocity_gradients);
+
+          MaterialModel::ElasticOutputs<dim>
+          *force_out = out.template get_additional_output<MaterialModel::ElasticOutputs<dim> >();
```

Comment:
Don't you define force_out in line 136 already?

Reply:
I believe the compiler complained when I did not redefine `force_out` a second time, but I'll check if defining it only once works.